persistence/src/main/scala/za/co/absa/spline/persistence/ArangoInit.scala

```diff
@@ -46,28 +50,39 @@ object ArangoInit extends ArangoInit {
-      _ <- db.createCollection("operation").toScala
-      _ <- db.createCollection("follows", new CollectionCreateOptions().`type`(CollectionType.EDGES)).toScala
-      _ <- db.createCollection("readsFrom", new CollectionCreateOptions().`type`(CollectionType.EDGES)).toScala
-      _ <- db.createCollection("writesTo", new CollectionCreateOptions().`type`(CollectionType.EDGES)).toScala
-      _ <- db.createCollection("dataSource").toScala
-      _ <- db.collection("dataSource").ensureHashIndex(Seq("uri").asJava, new HashIndexOptions().unique(true)).toScala
-      edgeDefs = Seq(
-        new EdgeDefinition().collection("progressOf").from("progress").to("execution"),
-        new EdgeDefinition().collection("executes").from("execution").to("operation"),
-        new EdgeDefinition().collection("follows").from("operation").to("operation"),
-        new EdgeDefinition().collection("readsFrom").from("operation").to("dataSource"),
-        new EdgeDefinition().collection("writesTo").from("operation").to("dataSource")
-      ).asJava
-      graph <- db.createGraph("lineage", edgeDefs).toScala
-    } yield graph
+  private def createDb(db: ArangoDatabaseAsync): Future[GraphEntity] = for {
+    _ <- db.create().toScala
+    _ <- createAQLUserFunctions(db)
+    _ <- db.createCollection("progress").toScala
+    _ <- db.createCollection("progressOf", new CollectionCreateOptions().`type`(CollectionType.EDGES)).toScala
+    _ <- db.createCollection("execution").toScala
+    _ <- db.createCollection("executes", new CollectionCreateOptions().`type`(CollectionType.EDGES)).toScala
+    _ <- db.createCollection("operation").toScala
+    _ <- db.createCollection("follows", new CollectionCreateOptions().`type`(CollectionType.EDGES)).toScala
+    _ <- db.createCollection("readsFrom", new CollectionCreateOptions().`type`(CollectionType.EDGES)).toScala
+    _ <- db.createCollection("writesTo", new CollectionCreateOptions().`type`(CollectionType.EDGES)).toScala
+    _ <- db.createCollection("dataSource").toScala
+    _ <- db.collection("dataSource").ensureHashIndex(Seq("uri").asJava, new HashIndexOptions().unique(true)).toScala
+    edgeDefs = Seq(
+      new EdgeDefinition().collection("progressOf").from("progress").to("execution"),
+      new EdgeDefinition().collection("executes").from("execution").to("operation"),
+      new EdgeDefinition().collection("follows").from("operation").to("operation"),
+      new EdgeDefinition().collection("readsFrom").from("operation").to("dataSource"),
+      new EdgeDefinition().collection("writesTo").from("operation").to("dataSource")
+    ).asJava
+    graph <- db.createGraph("lineage", edgeDefs).toScala
+  } yield graph
+
+  private def createAQLUserFunctions(db: ArangoDatabaseAsync) = {
+    val resolver = new PathMatchingResourcePatternResolver(getClass.getClassLoader)
+    val files: Array[Resource] = resolver.getResources("classpath:AQLFunctions/*.js")
+    files.map(file => {
```

Comment:
Prefer `foreach` to `map` when you execute it for a side effect, and the return value is ignored.